lib/ae_mdw/db/sync/block.ex

```diff
@@ -54,91 +57,76 @@ defmodule AeMdw.Db.Sync.Block do
 
     heights_hashes = Enum.reverse(heights_hashes)
 
-    heights_hashes
-    |> Enum.drop(if from_height == 0, do: 0, else: 1)
-    |> Enum.map(fn {height, hash} ->
-      kb_model = Model.block(index: {height, -1}, hash: hash, tx_index: 0)
-
-      WriteTxnMutation.new(Model.Block, kb_model)
-    end)
-    |> Database.commit()
-
     heights_hashes
     |> Enum.zip(Enum.drop(heights_hashes, 1))
-    |> Enum.each(fn {{height, kb_hash}, {_next_height, next_kb_hash}} ->
-      if rem(height, @log_freq) == 0, do: Log.info("syncing transactions at generation #{height}")
-
-      sync_generation(height, kb_hash, next_kb_hash)
+    |> Enum.reduce({[], from_txi}, fn {{height, kb_hash}, {_next_height, next_kb_hash}},
+                                      {blocks_mutations, txi} ->
+      {key_block, micro_blocks} = AE.Db.get_blocks(kb_hash, next_kb_hash)
+      kb_header = :aec_blocks.to_key_header(key_block)
+      kb_model = Model.block(index: {height, -1}, hash: kb_hash, tx_index: txi)
+
+      if rem(height, @log_freq) == 0, do: Log.info("creating mutations for block at #{height}")
+
+      pending_micro_blocks =
+        if from_height == height do
```

Comment:
`from_mbi != 0` maybe? (for semantics)

Reply:
If we don't check for `from_height  == height` then we would be skipping `from_mbi` blocks for every height

Reply:
At least for readability, looks like something can be improved, maybe putting both conditions:
`from_height == height and from_mbi != 0` but even after it the `from_height == height` part (from my perspective of not writing the code) would continue to be not easily readable.

Reply:
Done